### ISSUE TITLE
[rest_client] NoMethodError: undefined method `code' for nil:NilClass

### ISSUE THREAD

**Issue by frsantos** [DataDog/dd-trace-rb#636]
When invoking `RestClient::Request.execute` with a block that does not return the request.

For example:
``` ruby
RestClient::Request.execute(method: :get, url: 'https://www.google.com') {}

NoMethodError: undefined method `code' for nil:NilClass
from /.../gems/ruby-2.5.1@alkimii/gems/ddtrace-0.17.1/lib/ddtrace/contrib/rest_client/request_patch.rb:65:in `datadog_trace_request'
```

This is a case we faced when using Cloudinary. It uses `RestClient` [without returning a request](https://github.com/cloudinary/cloudinary_gem/blob/master/lib/cloudinary/uploader.rb#L327).

**Comment by delner** [comment]
Thanks for the report @frsantos! I'm going to look into this, see how to fix it up.

**Comment by delner** [comment]
Opened a pull request here: https://github.com/DataDog/dd-trace-rb/pull/642

**Comment by delner** [comment]
Released 0.17.3 which should resolve this.
